avocado/utils/pci.py

```
@@ -376,6 +376,91 @@
+
+        # unbind the device from its initial driver
+        cur_driver = get_driver(device)
+        if cur_driver is not None:
+            unbind(cur_driver, device)
+
+        # Bind device to driver
+        if driver is not None:
+            bind(driver, device)
+
+    except OSError as e:
+        raise ValueError(f"Not able to attach {driver} to {device}. Reason: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)